### PR TITLE
Add type annotations and enable mypy during ci

### PR DESCRIPTION
**Pull request by keisuke-umezawa**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -29,6 +29,19 @@ jobs:
...
+       - setup_remote_docker:
+          version: 18.05.0-ce
+          reusable: false
+       - run:
+           name: run mypy
+           command: |
+             make mypy
 
 workflows:
   version: 2
```
Comment (m3at) [inline .circleci/config.yml]: Need to change below to include mypy:
```yaml
   build:
     jobs:
       - black
       - test
       - mypy
```

File: src/backlight/datasource/__init__.py [hunk 2]
```diff
@@ -18,13 +22,15 @@ def load_marketdata(symbol, start_dt, end_dt, url):
     return from_dataframe(symbol, df)
 
 
-def from_dataframe(df, symbol, col_mapping=None):
+def from_dataframe(
+    df: pd.DataFrame, symbol: str, col_mapping: dict = None
```
Comment (m3at) [inline src/backlight/datasource/__init__.py]:
```suggestion
    df: pd.DataFrame, symbol: str, col_mapping: Optional[dict] = None
```
See [`typing.Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) for why, but basically `None` is not a `dict`.
You can add it to all subsequent values defaulting to `None`

File: src/backlight/labelizer/__init__.py [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
 from backlight.labelizer.common import LabelType
 from backlight.labelizer.labelizer import Labelizer, Label
+from backlight.query import query
 
 
-def load_label(symbol, url, start_dt=None, end_dt=None, mapping=None):
+def load_label(
+    symbol: str,
+    url: str,
+    start_dt: pd.Timestamp = None,
```
Comment (m3at) [inline src/backlight/labelizer/__init__.py]:
```suggestion
    start_dt: Optional[pd.Timestamp] = None,
```

File: src/backlight/labelizer/__init__.py [hunk 4]
```diff
@@ -1,22 +1,31 @@
...
+    end_dt: pd.Timestamp = None,
```
Comment (m3at) [inline src/backlight/labelizer/__init__.py]:
```suggestion
    end_dt: Optional[pd.Timestamp] = None,
```

File: src/backlight/labelizer/__init__.py [hunk 5]
```diff
@@ -1,22 +1,31 @@
...
+    mapping: dict = None,
```
Comment (m3at) [inline src/backlight/labelizer/__init__.py]:
```suggestion
    mapping: Optional[dict] = None,
```

File: src/backlight/labelizer/__init__.py [hunk 6]
```diff
@@ -1,22 +1,31 @@
...
+) -> Label:
     df = query(symbol, start_dt, end_dt, url)
     return from_dataframe(df=df, lbl_mapping=mapping)
 
 
-def from_dataframe(df, col_mapping=None, lbl_mapping=None):
+def from_dataframe(
+    df: pd.DataFrame, col_mapping: dict = None, lbl_mapping: dict = None
```
Comment (m3at) [inline src/backlight/labelizer/__init__.py]:
```suggestion
    df: pd.DataFrame, col_mapping: Optional[dict] = None, lbl_mapping: Optional[dict] = None
```

**Comment by BinrOp** [comment]
If it's useful, I often use this:
https://code.i-harness.com/en/q/da3b0f

Don't mind if irrelevant.

**Comment by m3at** [comment]
Thanks for adding type checking 👍

**Comment by BinrOp** [comment]
mypy tests pass! :tada:

**Comment by BinrOp** [comment]
I'll update the version of mypy for this repo too.

**Pull request merged**